capnp/src/private/arena.rs

```diff
@@ -159,7 +160,7 @@ pub trait BuilderArena: ReaderArena {
 /// A wrapper around a memory segment used in building a message.
 struct BuilderSegment {
     /// Pointer to the start of the segment.
-    ptr: *mut u8,
+    ptr: AtomicPtr<u8>,
```

Comment:
This looks scary, and it's probably unnecessary. It could be `*mut u8`,  and implement `Send`/`Sync` on `BuilderSegment` instead.